crates/valence_entity_hitbox/src/lib.rs

```diff
@@ -0,0 +1,523 @@
+#![doc = include_str!("../README.md")]
+#![allow(clippy::type_complexity)]
+
+use bevy_app::{App, CoreSet, Plugin};
+use bevy_ecs::prelude::{Component, Entity, SystemSet};
+use bevy_ecs::query::{Added, Changed, Or, With, WorldQuery};
+use bevy_ecs::schedule::{IntoSystemConfig, IntoSystemConfigs, IntoSystemSetConfig};
+use bevy_ecs::system::{Commands, Query};
+use glam::{DVec3, UVec3, Vec3Swizzles};
+use valence_core::aabb::Aabb;
+use valence_core::direction::Direction;
+use valence_entity::*;
+
+#[derive(SystemSet, Clone, Copy, Debug, Hash, PartialEq, Eq)]
+pub struct HitboxUpdateSet;
+
+#[cfg(feature = "hitbox_add")]
+#[derive(SystemSet, Clone, Copy, Debug, Hash, PartialEq, Eq)]
+pub struct HitboxComponentAddSet;
+
+pub struct HitboxPlugin;
+
+impl Plugin for HitboxPlugin {
+    fn build(&self, app: &mut App) {
+        app.configure_set(HitboxUpdateSet.in_base_set(CoreSet::PreUpdate))
+            .add_systems(
+                (
+                    update_constant_hitbox,
+                    update_warden_hitbox,
+                    update_area_effect_cloud_hitbox,
+                    update_armor_stand_hitbox,
+                    update_passive_child_hitbox,
+                    update_zombie_hitbox,
+                    update_piglin_hitbox,
+                    update_zoglin_hitbox,
+                    update_player_hitbox,
+                    update_item_frame_hitbox,
+                    update_slime_hitbox,
+                    update_painting_hitbox,
+                    update_shulker_hitbox,
+                )
+                    .in_set(HitboxUpdateSet),
+            );
+
+        #[cfg(feature = "hitbox_add")]
+        {
+            app.configure_set(HitboxComponentAddSet.in_base_set(CoreSet::PostUpdate))
+                .add_system(add_hitbox_component.in_set(HitboxComponentAddSet));
+        }
+    }
+}
+
+#[cfg(not(feature = "hitbox_add"))]
+type HitboxFilter<T> = Or<(T, Added<Hitbox>)>;
+
+#[cfg(not(feature = "hitbox_add"))]
+type HitboxFilterSingle<T> = Or<T, Added<Hitbox>>;
+
+#[cfg(feature = "hitbox_add")]
+type HitboxFilter<T> = Or<T>;
+
+#[cfg(feature = "hitbox_add")]
+type HitboxFilterSingle<T> = T;
+
+#[derive(Component, Debug)]
+pub struct Hitbox(Aabb);
```

Comment:
Might be better to derive `Component` on `Aabb` and use that directly instead of this newtype. A side effect of this is that users would be able to mutate the AABB, which might actually be desirable in some circumstances.

Reply:
But should they mutate it? Because it will be updated on each tick so it will just confuse some people and because of that bugs can occur